sky/templates/kubernetes-ray.yml.j2

```diff
@@ -322,18 +313,91 @@ available_node_types:
         - name: ray-node
           imagePullPolicy: IfNotPresent
           image: {{image_id}}
+          env:
+            - name: SKYPILOT_POD_NODE_TYPE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.labels['ray-node-type']
           # Do not change this command - it keeps the pod alive until it is
           # explicitly killed.
+          # We need to use -i to make sure the environment variables are set.
           command: ["/bin/bash", "-c", "--"]
           args: 
             - |
               # Helper function to conditionally use sudo
               prefix_cmd() { if [ $(id -u) -ne 0 ]; then echo "sudo"; else echo ""; fi; }
+              
+              (
+                {
+                mkdir -p ~/.sky
+                {{ conda_installation_commands }}
+                {{ ray_installation_commands }}
```

Comment:
I think conda depends on `curl` and if curl is not present in the image, this fails and the provision logs stay stuck at rsync stage before timing out. Maybe we need to:

1. Move the dependency installation to before ray installation
2. Have a better way to surface such failures, especially since rsync is run before we get to setup_runtime. Is there a way to write to container stdout only if these steps fail?

Repro: `sky launch -c test --cloud kubernetes --image-id docker:ubuntu:18.04 --num-nodes 1 --cpus 1 -- which python`

Reply:
Good point! Fixed in the latest commit, where we print the log out to container logs when it fails.